src/ProductConstructionService/ProductConstructionService.Common/MetricRecorder.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using Azure.Storage.Queues.Models;
+
+namespace ProductConstructionService.Common;
+
+public interface IMetricRecorder
+{
+    void QueueMessageReceived(QueueMessage message, TimeSpan delay);
+}
+
+public class MetricRecorder : IMetricRecorder
```

Comment:
Should this name be more explicit? Like `WorkItemWaitTime` recorder? We already have a TelemetryRecorder that's supposed to measure workitem durations and success rate, and other generic things.

Reply:
This will possibly record various different metrics

Reply:
We could merge this into the `TelemetryRecorder` but it would rather make sense to rename `TelemetryRecord` into `EvenRecorder`

Reply:
how should we differentiate it from the TelemetryRecorder, or do you see use merging those?

Reply:
`TelemetryRecorder` records events while this one records metrics so that's the difference. Either merging them or renaming `TelemetryRecorder` is possibly the way